packages/javascript/bh-shared-ui/src/components/ConfirmationDialog.tsx

```diff
@@ -14,49 +14,85 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-import { Button } from '@bloodhoundenterprise/doodleui';
-import { Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, FormHelperText } from '@mui/material';
-import React from 'react';
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogDescription,
+    DialogPortal,
+    DialogTitle,
+    Input,
+} from '@bloodhoundenterprise/doodleui';
+import { FormHelperText } from '@mui/material';
+import React, { useCallback, useState } from 'react';
 
 const ConfirmationDialog: React.FC<{
     open: boolean;
     title: string;
-    text: string;
+    text: string | JSX.Element;
     onClose: (response: boolean) => void;
+    challengeTxt?: string;
     isLoading?: boolean;
     error?: string;
-}> = ({ open, title, text, onClose, isLoading, error }) => {
+}> = ({ open, title, text, onClose, isLoading, error, challengeTxt }) => {
```

Comment:
```suggestion
}> = ({ open, title, text, onClose, isLoading, error, challengeTxt = '' }) => {
```
Defaulting to an empty string for this prop if it isn't passed can make the disabled confirm button logic a bit simpler: `isLoading || challengeTxt.toLowerCase() !== challengeTxtReply.toLowerCase()`